tests/test_jungfrau.py

```diff
@@ -1,22 +1,267 @@
+    elif axis==1:
+        assert(h5explorer.jungfrau1M.test_data.shape == (5, 2, 1024))
+    elif axis==2:
+        assert(h5explorer.jungfrau1M.test_data.shape == (5, 2, 512))
+    else:
+        assert(h5explorer.jungfrau1M.test_data.shape == (5, ))
+    logger.info('Correct shape')
+    
+    #checking for the type
+    for i in range(5):        
+        assert(h5explorer.jungfrau1M.test_data[i].dtype == 'float32')
+    logger.info('Correct type')
+    
+    logger.info('Pass the Projection_function test')
+    
+    tables.file._open_files.close_all()
+    
+    os.remove('test_proj.h5')
+
+
+@pytest.mark.parametrize('getROIs', [{'ROI': [[1,2], [157,487], [294,598]], 'writeArea': True, 'thresADU': None, }], indirect=True)
+@pytest.mark.parametrize('getProjections', [{'axis': 1, 'mean': True, 'thresADU': 1e-6, 'thresRms': 1e-6, 'singlePhoton': False}], indirect=True)
+@pytest.mark.parametrize('datasource', [{'exp': 'xpptut15', 'run': 650}], indirect=True)
+@pytest.mark.parametrize('detector', [{'name': 'jungfrau1M'}], indirect=True)
+def test_ROI_Projection(getROIs, getProjections, detector, datasource):
+    logger.info('Running test for ROI_Projection')
+    #roi function
+    try:
+        func_kwargs = getROIs
+    except Exception as e:
+        print(f'Can\'t instantiate ROI args: {e}')
+        func_kwargs = []
+    writeArea = func_kwargs['writeArea']
+    [[x1, x2], [y1, y2], [z1, z2]] = func_kwargs['ROI']
+    func = ROIFunc(**func_kwargs)
+    #project function
+    try:
+        func_kwargs = getProjections
+    except Exception as e:
+        print(f'Can\'t instantiate Projection args: {e}')
+        func_kwargs = []
+    axis = func_kwargs['axis']
+    proj = projectionFunc(**func_kwargs)
+    
+    if axis >= 2:
+        print('axis out of bound')
+        logger.info('Test failed because of wrong axis')
+        return
+    
+    func.addFunc(proj)
+    
+    det = detector
+    ds, run = datasource
+    det.addFunc(func)
+    
+    # Process a couple events
+    userDict = {} # dictionary to store detector data
+    small_data = ds.small_data('./test_roi_proj.h5', gather_interval=5) # file to save data to
+    ds.break_after(5) # stop event loop after 5 events
+
+    for nevt,evt in enumerate(ds.events()): # usual psana event loop
+        det.getData(evt) # get the detector data
+        det.processFuncs() # process the attached functions
+        userDict[det._name]=getUserData(det) # get the function results
+        small_data.event(userDict) # write data to h5
+    
+    # Test the function
+    f = "test_roi_proj.h5"
+    h5explorer = tables.File(f).root
+    
+    #checking for the output
+    if writeArea:
+        assert(list(h5explorer.jungfrau1M._v_children)==['ROI_area', 'ROI_com', 'ROI_max', 'ROI_mean', 'ROI_sum', 'ROI_test_data'])
+    else:
+        assert(list(h5explorer.jungfrau1M._v_children)==['ROI_com', 'ROI_max', 'ROI_mean', 'ROI_sum', 'ROI_test_data'])
+    logger.info('Correct output')
```

Comment:
There should also be some output for the projection there. Let's add them as well to the list and test them as for the other entries (type and shape)